File: problem/Unique_Number_Occurrences/uniqueOccurrences.java
import java.util.*;

class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        Arrays.sort(arr);
        List<Integer> counts = new ArrayList<>();
        int i = 0;
        while (i < arr.length) {
            int cnt = 1;
            while (i + 1 < arr.length && arr[i] == arr[i + 1]) {
                cnt++;
                i++;
            }
            counts.add(cnt);
            i++;
        }
        Collections.sort(counts);
        for (int j = 1; j < counts.size(); j++) {
            if (Objects.equals(counts.get(j), counts.get(j - 1))) return false;
        }
        return true;
    }
}

File: problem/Unique_Number_Occurrences/uniqueOccurrences.py
class Solution:
    def uniqueOccurrences(self, arr):
        arr.sort()
        counts = []
        i = 0
        n = len(arr)
        while i < n:
            cnt = 1
            while i + 1 < n and arr[i] == arr[i + 1]:
                cnt += 1
                i += 1
            counts.append(cnt)
            i += 1
        counts.sort()
        for j in range(1, len(counts)):
            if counts[j] == counts[j - 1]:
                return False
        return True
